fem/fespace.hpp

```diff
@@ -107,25 +107,49 @@ class FiniteElementSpace
    /// Number of degrees of freedom. Number of unknowns is #ndofs * #vdim.
    int ndofs;
 
-   int nvdofs, nedofs, nfdofs, nbdofs;
-   int *fdofs, *bdofs;
+   /** Polynomial order for each element. If empty, all elements are assumed
+       to be of the default order (fec->GetOrder()). */
+   Array<char> elem_order;
 
-   mutable Table *elem_dof; // if NURBS FE space, not owned; otherwise, owned.
-   mutable Table *bdrElem_dof; // not owned only if NURBS FE space.
-   mutable Table *face_dof; // owned
-   mutable Array<int> face_to_be; // used only with NURBS FE spaces; owned.
+   int nvdofs, nedofs, nfdofs, nbdofs;
+   int uni_fdof; ///< # of single face DOFs if all faces uniform; -1 otherwise
+   int *bdofs; ///< internal DOFs of elements if mixed/var-order; NULL otherwise
+
+   /** Variable-order spaces only: these tables hold DOF assignments for edges
+       and faces, each of which in a variable order space may host several DOF
+       sets, called DOF set variants. Example: an edge 'i' shared by 4 hexes of
+       orders 2, 3, 4, 5 will hold four DOF sets starting at indices e.g. 100,
+       101, 103, 106. These numbers are stored in row 'i' of \a var_edge_dofs.
+       Variant zero is always the lowest order DOF set, which is followed by
```

Comment:
I am more confused with this example than without it. Are the indices 100, 101, 103, 106 determined by the orders, or are they just randomly chosen numbers that might be the indices?

Reply:
These are examples of DOF numbers, as in "this 5th-order edge has internal DOFs 100-103". I didn't feel that users need to study this in detail but Yohann requested a documentation for these tables, so I added one. If it's more confusing than useful I can remove the comment, what do you think @tzanio?

Reply:
This is definitely for developers, not users. Could there be a better place, maybe somewhere in the implementation, where this can be moved and expanded? I believe @YohannDudouit's request was for information that can help MFEM developers (e.g. folks at LLNL) that wants to understand the algorithm and what basic data structures are involved...